### PR TITLE
[Core] Remove Legacy Raylet Code

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

Remove all raylet legacy code after GCS actor management is on.

```
Remove gcs_actor_service_enabled (o)
Check actor_creator logic (o)
Remove HandleDisconnectActor (o)
Remove ResubmitTask (o)
Remove HandleTaskReconstruction (o)
Remove raylet transport (o)
Remove actor reporter (o)
FinishAssignedActorCreationTask (o)
FinishAssignedActorTask (Still need it)
Remove IsActorTask() if statement (o)
No need for lineage_cache (o)
HandleForwardTask (o)
Remove ActorFrontier (o)
```

TODO
- Remove dummy object ID
- Modify method names for `ForwardTask` or `AssignTask`, which are not accurate description anymore. <= should be done by new scheduler refactoring.
- Reconstruction policy class has one path that is still used, which is to mark object failed when the task lease is expired. We should probably refactor the class or find a simplified logic.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: src/ray/raylet/format/node_manager.fbs [hunk 1]
```diff
@@ -69,8 +69,6 @@ enum MessageType:int {
   // before the checkpoint, and make any tasks on the frontier runnable by
   // making their execution dependencies available.
   SetActorFrontier,
```
Comment (ericl) [inline src/ray/raylet/format/node_manager.fbs:71]: Can we also remove the ActorFrontier stuff above?

**Review by ericl — APPROVED** [review]

**Review by edoakes — APPROVED** [review]
-3,793 LOC :D

File: src/ray/raylet/format/node_manager.fbs (hunk 1 above)
Comment (rkooo567) [reply src/ray/raylet/format/node_manager.fbs]: Removed `UpdateActorFrontier` (method that doesn't have cc body), `SetActorFrontier`, `GetActorFrontierReply`, `GetActorFrontierRequest` (flat buffer)

**Comment by rkooo567** [comment]
@ericl @edoakes I had to revert removing reconstruction policy because there's one code path that is still used. We should mark object as failed when task lease is expired (without this wait on object that is LRU evicted hanging forever (e.g., running `test_advanced_3.py::test_wait_reconstruction ` with `object_pinning_enabled: 0` hangs forever)). I will create a separate PR to address this later (after discussing with Stephanie the right solution).

**Pull request merged**